wandb/apis/public/runs.py

```diff
@@ -161,6 +167,80 @@ def convert_objects(self):
 
         return objs
 
+    @normalize_exceptions
+    def histories(
+        self,
+        samples: int = 500,
+        keys: list = None,
+        x_axis: str = "_step",
+        format: Literal["default", "pandas", "polars"] = "default",
+        stream: Literal["default", "system"] = "default",
+    ):
+        """Return sampled history metrics for all runs that fit the filters conditions.
+
+        Arguments:
+            samples : (int, optional) The number of samples to return per run
+            keys : (list, optional) Only return metrics for specific keys
+            x_axis : (str, optional) Use this metric as the xAxis defaults to _step
+            format : (Literal, optional) Format to return data in, options are "default", "pandas", "polars"
+            stream : (Literal, optional) "default" for metrics, "system" for machine metrics
+        Returns:
+            pandas.DataFrame: If format="pandas", returns a `pandas.DataFrame` of history metrics indexed by run id.
+            polars.DataFrame: If format="polars", returns a `polars.DataFrame` of history metrics indexed by run id.
+            list of dicts: If format="default", returns a list of dicts containing history metrics with a run_id key.
+        """
+        all_histories = []
+
+        pd = None
+        pl = None
+
+        if format == "pandas":
+            pd = util.get_module(
+                "pandas", required="Logging dataframes requires pandas"
+            )
+        elif format == "polars":
+            pd = util.get_module(
+                "pandas", required="Exporting Polars DataFrame requires pandas"
```

Comment:
Currently, pandas is required to return Polars dataframes. This implementation was chosen to avoid modifying the `run.history` method, which takes a boolean argument pandas. Now, the check is performed in the wrapper `runs.histories` using `pandas=(format in ["pandas", "polars"])`. 

We could remove the pandas requirement by changing the history method. Please let me know if this is preferred; I can adapt it accordingly, though this may break existing code that relies on `run.history(pandas)`.

Reply:
it's not needed -- see my fix